titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/internal/JobFeatureComplianceChecks.java

```diff
@@ -106,6 +111,39 @@
         };
     }
 
+    /**
+     * A feature compliance is violated if there is a default accountId and subnets combination present in the {@link JobManagerConfiguration} for the deployment stack
+     * but the container attributes in the {@link JobDescriptor} are missing one or both values.
+     * @param jobManagerConfiguration {@link JobManagerConfiguration}
+     * @return feature compliance evaluation
+     */
+    static FeatureCompliance<JobDescriptor<?>> missingContainerAccountIdAndSubnets(JobManagerConfiguration jobManagerConfiguration) {
+        return jobDescriptor -> {
+            String defaultAccountId = jobManagerConfiguration.getDefaultContainerAccountId();
+            String defaultSubnets = jobManagerConfiguration.getDefaultSubnets();
+            // Ignore this compliance check unless both default accountId and subnet configuration properties are set implying our
+            // intent to aid the job descriptor sanitization
+            if (StringExt.isEmpty(defaultAccountId) || StringExt.isEmpty(defaultSubnets)) {
+                return Optional.empty();
+            }
+
+            String accountIdContainerAttribute = jobDescriptor.getContainer().getAttributes().get(JOB_CONTAINER_ATTRIBUTE_ACCOUNT_ID);
+            String subnetContainerAttribute = jobDescriptor.getContainer().getAttributes().get(JOB_CONTAINER_ATTRIBUTE_SUBNETS);
+
+            // Feature compliance is violated if either (i) both attributes are not set, or (ii) accountId is set to the default value but the subnets value is not set.
+            // In either case, we should take action in response to this violation.
+            if ((StringExt.isEmpty(accountIdContainerAttribute) || defaultAccountId.equals(accountIdContainerAttribute)) && StringExt.isEmpty(subnetContainerAttribute)) {
```

Comment:
If I understand it correctly, setting accontId parameter to the default account id is non-compliant?

Reply:
Non compliant if:
(1). default account_id and subnets are set in the configuration (expected for the stack),
AND
(2a). account_id and subnets are empty in the job descriptor, or
(2b). account_id is set to the default account_id but subnets is not set in the job descriptor

All other cases are given a pass under current/default behavior.